src/segments/golang.go

```diff
@@ -11,15 +12,16 @@ type Golang struct {
 }
 
 const (
-	ParseModFile properties.Property = "parse_mod_file"
+	ParseModFile    properties.Property = "parse_mod_file"
+	ParseGoWorkFile properties.Property = "parse_go_work_file"
 )
 
 func (g *Golang) Template() string {
 	return languageTemplate
 }
 
 func (g *Golang) Enabled() bool {
-	g.extensions = []string{"*.go", "go.mod"}
+	g.extensions = []string{"*.go", "go.mod", "go.sum", "go.work"}
```

Comment:
Why do we need to add go.sum? Should we also add go.work.sum?

Reply:
If they are present, it's likely to be a Go project. We could definitely add the go.work.sum too